### ISSUE TITLE
the model manage to get parameters by getParameters() but fail to perform update parameters by updateParameter()

### ISSUE THREAD

**Issue by yjy765** [torch/nn#868]
hello everyone
I try to use customized layer which contains some shared network
The real one I made is so complex to test so I made simple layer composed of shared linear network as follows

`local test, parent = torch.class('nn.test','nn.Module')

 function test:__init()
   parent.__init(self)
   self.node = nn.Sequential():add(nn.Linear(3,2))
   self.module = {}
   self.module[1] = self.node
end

function test:get(step)
  local module = self.module[step]
  if not module then
    module = self.node:clone('weight','bias','gradWeight','gradBias'
    self.module[step] = module
  end
  return module
end

function test:updateOutput(input)
  self.output = {}
  for i=1,3 do
    self.output[i] = self:get(i):forward(input)
  end
  return self.output
end

function test:updateGradInput(input,gradOutput)
  self.gradInput = torch.Tensor(input:size()):zero()
  for i=1,3 do
    self.gradInput = self.gradInput + self:get(i):backward(input,gradOutput[i])
  end
  return self.gradInput
end

function test:accGradParameters(input,gradOutput,scale)
  for i=1,3 do
    self:get(i):zeroGradParameters()
    self:get(i):accGradParameters(input,gradOutput[i],scale)
  end
end`

(To shortly sum up, the layer contains three linear layers in parallel which share all parameters each other)
so I made a very simple main script as like

input = nn.Identity()()
output = nn.test()(input)
model = nn.gModule(input,output)

then the forward and backward of model work well

However when I execute model:updateParameters(lr), there is no change of parameters

When I call model:getParameters() then it returns 8 parameters and 8 gradient parameters

Also if I do model:zeroGradParameters(), again no change of gradient parameters. Hence, I added     self:get(i):zeroGradParameters() in the accGradParameters function.

One more strange thing is 'model:getParameters()' works successfully but it does not return anything when I call 'model:parameters()'. In the Module.lua file, as I understand, 'model:parameter()' should return the same thing with 'model:getParameters()'.

Can anyone explain and solve the problem plz?

(The composition and idea of the code are all borrowed from Element-Research/rnn)
